core/templates/domain/skill/editable-skill-backend-api.service.ts

```diff
@@ -16,124 +16,159 @@
-          response.data.grouped_skill_summaries);
-        if (successCallback) {
-          successCallback({
-            skill: skill,
-            groupedSkillSummaries: groupedSkillSummaryDicts
-          });
-        }
-      }, function(errorResponse) {
+// TODO(#7165): Replace any with exact type.
+interface editableSkillResponseConfig {
+    skill?: any;
+    skills?: any;
+    // eslint-disable-next-line camelcase
+    grouped_skill_summaries?: any;
+}
+@Injectable({
+  providedIn: 'root'
+})
+export class EditableSkillBackendApiService {
+  constructor(
+    private http: HttpClient,
+    private urlInterpolationService: UrlInterpolationService
+  ) {}
+
+  _fetchSkill(
+      skillId: string,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void
+  ): void {
+    const skillDataUrl = this.urlInterpolationService.interpolateUrl(
+      SkillDomainConstants.EDITABLE_SKILL_DATA_URL_TEMPLATE, {
+        skill_id: skillId
+      }
+    );
+    this.http.get(skillDataUrl).toPromise().then(
+      (data: editableSkillResponseConfig) => successCallback({
+        skill: data.skill,
+        groupedSkillSummaries: data.grouped_skill_summaries
+      }),
+      error => {
         if (errorCallback) {
-          errorCallback(errorResponse.data);
+          errorCallback(error.error);
         }
-      });
-    };
-
-    var _fetchMultiSkills = function(skillIds, successCallback, errorCallback) {
-      var skillDataUrl = UrlInterpolationService.interpolateUrl(
-        SKILL_DATA_URL_TEMPLATE, {
-          comma_separated_skill_ids: skillIds.join(',')
-        });
+      }
+    );
+  }
 
-      $http.get(skillDataUrl).then(function(response) {
-        var skills = angular.copy(response.data.skills);
-        if (successCallback) {
-          successCallback(skills);
-        }
-      }, function(errorResponse) {
+  _fetchMultiSkills(
+      skillIds: [string],
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void
+  ): void {
+    const skillDataUrl = this.urlInterpolationService.interpolateUrl(
+      SkillDomainConstants.SKILL_DATA_URL_TEMPLATE, {
+        comma_separated_skill_ids: skillIds.join(',')
+      }
+    );
+    this.http.get(skillDataUrl).toPromise().then(
+      (data: editableSkillResponseConfig) => successCallback(data.skills),
+      error => {
         if (errorCallback) {
-          errorCallback(errorResponse.data);
+          errorCallback(error.error);
         }
-      });
-    };
-
-    var _updateSkill = function(
-        skillId, skillVersion, commitMessage, changeList,
-        successCallback, errorCallback) {
-      var editableSkillDataUrl = UrlInterpolationService.interpolateUrl(
-        EDITABLE_SKILL_DATA_URL_TEMPLATE, {
-          skill_id: skillId
-        });
+      }
+    );
+  }
+  // TODO(#7165): Replace any with exact type.
```

Comment:
skillVersion would be string right? Could you please confirm?

Also, move some params to the same line. We don't expect each param in different line. Ditto here and below.

Reply:
skillVersion is a number

Reply:
I will make the changes

Reply:
Done!